bigc/api_client.py

```diff
@@ -13,20 +11,44 @@
 
 
 class BigCommerceRequestClient(ABC):
-    def __init__(self, store_hash: str, access_token: str, timeout: Optional[float] = None):
+    def __init__(self, store_hash: str, access_token: str, timeout: float | None = None):
         self.store_hash = store_hash
         self.access_token = access_token
         self.timeout = timeout
 
-    def request(self, method: str, path: str, **kwargs):
+    def request(
+        self,
+        method: str,
+        path: str,
+        *,
+        data: Any = None,
+        json: Any = None,
```

Comment:
Not sure if it makes sense to include both `data` and `json`, since one of the assumptions of this class is that all requests/responses are JSON-encoded. I would probably just leave the `data` arg and map it to the `json` argument in `requests.request()`.